File: CookOff_2018_08_BFTT.py
# code_report Solution
# https://youtu.be/Enbht_Uf1CE

t = int (input ())
for i in range (t):
    n = int (input ()) + 1
    while (str (n).count ('3') < 3):
        n = n + 1
    print (n)
    

File: CookOff_2018_08_BFTT.java
// code_report Solution
// https://youtu.be/Enbht_Uf1CE

import java.util.*;
import java.io.*;

public class Main {
     public static void main (String []args) {
         Scanner sc = new Scanner (System.in);
         int t = sc.nextInt ();
         while (t-- != 0) {
             int n = sc.nextInt ();
             while (true) {
                n++;
                String s = Integer.toString (n); 
                long   c = s.chars ().filter (ch -> ch =='3').count ();
                if (c < 3) continue;
                System.out.println (n);
                break;
             }
         }
     }
}
